src/release/breaking-changes/android-predictive-back.md

```diff
@@ -0,0 +1,314 @@
+In many cases, logic that was being run at the time of the back gesture in
+`onWillPop` can be done at build time and set to `canPop`.
+
+Code before migration:
+
+```dart
+WillPopScope(
+  onWillPop: () async {
+    return _myCondition;
+  },
+  child: ...
+),
+```
+
+Code after migration:
+
+```dart
+PopScope(
+  canPop: _myCondition,
+  child: ...
+),
+```
+
+For cases where it's necessary to be notified that a pop was attempted, the
+`onPopInvoked` method can be used in a similar way to `onWillPop`. Keep in mind
+that while `onWillPop` was called before the pop was handled and had the ability
+to cancel it, `onPopInvoked` is called after the pop is finished being handled.
+
+Code before migration:
+
+```dart
+WillPopScope(
+  onWillPop: () async {
+    _myHandleOnPopMethod();
+    return true;
+  },
+  child: ...
+),
+```
+
+Code after migration:
+
+```dart
+PopScope(
+  canPop: true,
+  onPopInvoked (bool didPop) {
+    _myHandleOnPopMethod();
+  },
+  child: ...
+),
+```
+
+### Migrating from `WillPopScope` to `NavigatorPopHandler` for nested `Navigator`s
+A very common use case of `WillPopScope` was to properly handle back gestures
+when using nested `Navigator` widgets. It's possible to do this using `PopScope`
+as well, but there is now a wrapper widget that makes this even easier:
+`NavigatorPopHandler`.
+
+Code before migration:
+
+```dart
+WillPopScope(
+  onWillPop: () async => !(await _nestedNavigatorKey.currentState!.maybePop()),
+  child: Navigator(
+    key: _nestedNavigatorKey,
+    …
+  ),
+)
+```
+
+Code after migration:
+
+```dart
+NavigatorPopHandler(
+  onPop: () => _nestedNavigatorKey.currentState!.pop(),
+  child: Navigator(
+    key: _nestedNavigatorKey,
+    …
+  ),
+)
+```
+
+### Migrating from `Form.onWillPop` to `Form.canPop` and `Form.onPopInvoked`
+`Form` used to use a `WillPopScope` under the hood and expose its `onWillPop`
+method. It has been replaced with a `PopScope` and has exposed its `canPop` and
+`onPopInvoked` methods. Migrating is identical to migrating from `WillPopScope`
+to `PopScope`, detailed above.
+
+### Migrating from `Route.willPop` to `Route.popDisposition`
+`Route`'s `willPop` method returned a `Future<RoutePopDisposition>` to
+accommodate the fact that pops could be canceled.  Now that that's no longer
+true, this logic has been simplified to a synchronous getter.
+
+Code before migration:
+
+```dart
+if (await myRoute.willPop() == RoutePopDisposition.doNotPop) {
+  ...
+}
+```
+
+Code after migration:
+
+```dart
+if (myRoute.popDisposition == RoutePopDisposition.doNotPop) {
+  ...
+}
+```
+
+### Migrating from `ModalRoute.add/removeScopedWillPopCallback` to `ModalRoute.(un)registerPopInterface`
+Internally, `ModalRoute` kept track of the existence of `WillPopScope`s in its
+widget subtree by registering them with `addScopedWillPopCallback` and
+`removeScopedWillPopCallback`. Since `WillPopScope` has been replaced by
+`PopScope`, these methods have been replaced by `registerPopInterface` and
```

Comment:
```suggestion
`removeScopedWillPopCallback`. Since `PopScope` replaces `WillPopScope`,
these methods have been replaced by `registerPopInterface` and
```